prow/pjutil/abort.go

```diff
@@ -0,0 +1,75 @@
+package pjutil
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+// prowClient a minimalistic prow client required by the aborter
+type prowClient interface {
+	//ReplaceProwJob updates the prow job
+	ReplaceProwJob(string, prowapi.ProwJob) (prowapi.ProwJob, error)
+}
+
+// ProwJobResourcesCleanupFn type for a callback function which it is expected to clean up
+// all k8s resources associated with the given prow job
+type ProwJobResourcesCleanupFn func(pj prowapi.ProwJob) error
+
+// ProwJobAborter provides functionality to abort prow jobs
+type ProwJobAborter struct {
```

Comment:
Shouldn't we export the interface, not the impl?

Reply:
I defined an interface.